File: app/src/main/java/com/natateam/myevents/mainlist/MainActivityComponent.java
package com.natateam.myevents.mainlist;

import com.natateam.myevents.MainActivity;

import dagger.Component;
import dagger.Subcomponent;
import dagger.android.AndroidInjector;

/**
 * Created by macbook on 29/11/ 15.
 */
@Subcomponent(modules = {MainActivityModule.class})
public interface MainActivityComponent extends AndroidInjector<MainActivity> {
    @Subcomponent.Builder
    abstract class Builder extends AndroidInjector.Builder<MainActivity>{}
}

File: app/src/main/java/com/natateam/myevents/mainlist/MainActivityComponent.kt
package com.natateam.myevents.mainlist

import com.natateam.myevents.MainActivity
import com.natateam.myevents.mainlist.fragments.BirthdaysFragment
import com.natateam.myevents.mainlist.fragments.RepeatFragment
import com.natateam.myevents.mainlist.fragments.TodoFragment

import dagger.Component
import dagger.Subcomponent
import dagger.android.AndroidInjector

/**
 * Created by macbook on 29/11/ 15.
 */
@Subcomponent(modules = arrayOf(EventListModule::class))
interface MainActivityComponent  {
    fun injectTo(view: TodoFragment)
    fun injectTo(view: RepeatFragment)


    @Subcomponent.Builder
     interface Builder {
        fun eventListModule(eventListModule: EventListModule): Builder
        fun build(): MainActivityComponent
    }
}
